parsing/location.ml

```diff
@@ -225,6 +225,89 @@ let print_locs ppf locs =
   Format.pp_print_list ~pp_sep:(fun ppf () -> Format.fprintf ppf ",@ ")
     print_loc ppf locs
 
+(******************************************************************************)
+(* An interval set structure; additionally, it stores user-provided information
+   at interval boundaries.
+
+   The implementation provided here is naive and assumes the number of intervals
+   to be small, but the interface would allow for a more efficient
+   implementation if needed.
+
+   Note: the structure only stores maximal intervals (that therefore do not
+   overlap).
+*)
+
+module IMapSet : sig
+  type 'a t
+  (* bounds are included *)
+  val of_intervals : (('a * int) * ('a * int)) list -> 'a t
+
+  val mem : 'a t -> pos:int -> bool
+  val find_in : 'a t -> s:int -> e:int -> 'a option
```

Comment:
At first I couldn't tell what `s` and `e` were for here. Maybe just `int * int`, or more explicit label names.

The specification, if I understand the implementation correctly, is: return the value associated to one of the interval bounds that lies within the `(s, e)` parameters. Maybe `find_bound_in`?

Reply:
Should be better now (`val find_bound_in : 'a t -> range:(int * int) -> 'a bound option`).